src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/adu_communication_channel.c

```diff
@@ -0,0 +1,1003 @@
+ * client. */
+void ADUC_CommunicationChannel_OnDisconnect(struct mosquitto* mosq, void* obj, int rc, const mosquitto_property* props)
+{
+    Log_Info("on_disconnect: reason=%s", mosquitto_strerror(rc));
+    ADUC_SetCommunicationChannelState(ADU_COMMUNICATION_CHANNEL_CONNECTION_STATE_DISCONNECTED);
+
+    // TODO (nox-msft) - select retry strategy based on error code.
+    //
+    // ADUC_MQTT_DISCONNECTION_CATEGORY category = CategorizeMQTTDisconnection(rc);
+    // switch (category)
+    // {
+    // case ADUC_MQTT_DISCONNECTION_CATEGORY_TRANSIENT:
+    //     break;
+    // case ADUC_MQTT_DISCONNECTION_CATEGORY_NON_RECOVERABLE:
+    //     // This may require the application restart
+    //     break;
+    // case ADUC_MQTT_DISCONNECTION_CATEGORY_OTHER:
+    // default:
+    //     break;
+    // }
+    //
+    // For now, wait 30 seconds before retrying.
+    s_commMgrState.commNextRetryTime = ADUC_GetTimeSinceEpochInSeconds() + DEFAULT_CONNECT_RETRY_DELAY_SECONDS;
+
+    ADUC_SetCommunicationChannelState(ADU_COMMUNICATION_CHANNEL_CONNECTION_STATE_DISCONNECTED);
+
+    if (s_commMgrState.mqttCallbacks.on_disconnect_v5 != NULL)
+    {
+        s_commMgrState.mqttCallbacks.on_disconnect_v5(mosq, obj, rc, props);
+    }
+}
+
+/*
+ * @brief  Initialize the communication channel.
+ * @param[in] session_id The session ID to use for the MQTT client.
+ * @param[in] module_state The module state to use for the MQTT client.
+ * @param[in] callbacks The MQTT callbacks to use for the MQTT client.
+ * @return true if the communication channel was successfully initialized; otherwise, false.
+ */
+bool ADUC_CommunicationChannel_Initialize(const char* session_id, void* module_state, ADUC_MQTT_CALLBACKS* callbacks)
+{
+    int mqtt_res = 0;
+    bool use_OS_cert = false;
+
+    if (s_commMgrState.initialized)
+    {
+        return true;
+    }
+
+    // Copy all callbacks address.
+    memcpy(&s_commMgrState.mqttCallbacks, callbacks, sizeof(ADUC_MQTT_CALLBACKS));
+
+    if (!ReadMqttBrokerSettings(&s_commMgrState.mqttSettings))
+    {
+        Log_Error("Failed to read Azure DPS2 MQTT settings");
+        goto done;
+    }
+
+    if (!Prepare_ADU_MQTT_Topics())
+    {
+        goto done;
+    }
+
+    Log_Info("Initialize Mosquitto library");
+    mosquitto_lib_init();
+
+    s_commMgrState.mqttClient = mosquitto_new(session_id, s_commMgrState.mqttSettings.cleanSession, module_state);
+    if (!s_commMgrState.mqttClient)
+    {
+        Log_Error("Failed to create Mosquitto client");
+        return -1; // Initialization failed
+    }
+
+    mqtt_res = mosquitto_int_option(
+        s_commMgrState.mqttClient, MOSQ_OPT_PROTOCOL_VERSION, s_commMgrState.mqttSettings.mqttVersion);
+    if (mqtt_res != MOSQ_ERR_SUCCESS)
+    {
+        Log_Error("Failed to set MQTT protocol version (%d)", mqtt_res);
+        goto done;
+    }
+
+    if (!IsNullOrEmpty(s_commMgrState.mqttSettings.username))
```

Comment:
username + password unused. Let's remove these unless there is compelling reason to keep

Reply:
Actually, username is required for both DPS and EG MQTT Broker. I can remove password.

Reply:
yes, please remove password at very least since it's not needed.